xla/service/gpu/thunk.cc

```diff
@@ -225,9 +232,21 @@ Thunk::ExecuteParams::ExecuteParams(
     CASE(kTriangularSolve);
     CASE(kWhile);
     CASE(kFusedMHA);
+    CASE(kSyncOnStreams);
   }
 }
 
+/*static*/
+se::Stream* Thunk::GetStreamForExecution(ExecutionStreamId stream_id,
+                                         const ExecuteParams& params) {
+  if (stream_id == GetMainComputeStreamId()) {
+    return params.stream;
+  }
+  auto iter = params.additional_compute_streams.find(stream_id);
+  CHECK(iter != params.additional_compute_streams.end());
```

Comment:
new checks are strongly discouraged in XLA, `absl::StatusOr<se::Stream*>` is preferred result type here

Reply:
i changed it to return an invalidArg error instead of a CHECK.